calyx-py/test/correctness/queues/binheap/stable_binheap.py

```diff
@@ -0,0 +1,177 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+from binheap import insert_binheap
+
+
+def insert_stable_binheap(prog, name, queue_len_factor):
+    """Inserts the component `insertion_order_binheap` into the program.
+
+    It is a minimum binary heap that breaks ties via insertion order.
+
+    It has:
+    - three inputs, `cmd`, `value`, and `rank`.
+    - two ref registers, `ans` and `err`.
+    """
+
+    comp = prog.component(name)
+
+    below = comp.cell("below", insert_binheap(prog, "below", queue_len_factor))
+
+    cmd = comp.input("cmd", 2)
+
+    rank = comp.input("rank", 32)
+    value = comp.input("value", 32)
+
+    ans = comp.reg(32, "ans", is_ref=True)
+
+    err = comp.reg(1, "err", is_ref=True)
+
+    i = comp.reg(32)
+
+    cat = comp.cat(32, 32)
+
+    with comp.continuous:
+        cat.left = rank
+        cat.right = i.out
+
+    comp.control += [
+        cb.invoke(
+            below, 
+            in_value=value, 
+            in_rank=cat.out, 
+            in_cmd=cmd, 
+            ref_ans=ans, 
+            ref_err=err
+        ),
+        comp.incr(i)
+    ]
```

Comment:
One more place you'll want to explain the strategy

Reply:
Done in [de13dc7](https://github.com/calyxir/calyx/pull/2164/commits/de13dc7e0ea656d8e452d19f98651e75c05c13f9)